tsl/test/isolation/specs/deadlock_drop_index_vacuum.spec

```diff
@@ -0,0 +1,124 @@
+# This file and its contents are licensed under the Timescale License.
+# Please see the included NOTICE for copyright information and
+# LICENSE-TIMESCALE for a copy of the license.
+
+# Order of locks in drop index and vacuum analyze was wrong, and DROP
+# INDEX took the locks in order index-table, while VACUUM ANALYZE took
+# them in order table-index.
+#
+# Create deadlock if the locking order is wrong. Problem here is that
+# vacuum takes two locks. First one to read a table and then one to do
+# the actual vacuum, so we need two sessions working in lock-step to
+# create the deadlock.
+# 
+# S1: Lock chunk from hypertable
+#      index:    []
+#      table: S1 []
```

Comment:
Great illustration, but can you clarify what the "symbols" mean? It is not obvious.

After some thinking, I think the brackets indicate lock queue, or?

Reply:
> Great illustration, but can you clarify what the "symbols" mean? It is not obvious.
> 
> After some thinking, I think the brackets indicate lock queue, or?

Yes, that is the lock queue.

Reply:
Added a legend as a paragraph before the sequence.